optimum/habana/transformers/models/mt5/modeling_mt5.py

```diff
@@ -0,0 +1,662 @@
+            mask = torch.ones(position_bias.shape[1])
+            mask[list(self.pruned_heads)] = 0
+            position_bias_masked = position_bias[:, mask.bool()]
+        else:
+            position_bias_masked = position_bias
+
+        scores += position_bias_masked
+        attn_weights = nn.functional.softmax(scores.float(), dim=-1).type_as(
+            scores
+        )  # (batch_size, n_heads, seq_length, key_length)
+        if self.training:
+            htcore.mark_step()
+        attn_weights = nn.functional.dropout(
+            attn_weights, p=self.dropout, training=self.training
+        )  # (batch_size, n_heads, seq_length, key_length)
+        if self.training:
+            htcore.mark_step()
+
+        # Mask heads if we want to
+        if layer_head_mask is not None:
+            attn_weights = attn_weights * layer_head_mask
+
+        attn_output = unshape(torch.matmul(attn_weights, value_states))  # (batch_size, seq_length, dim)
+        attn_output = self.o(attn_output)
+
+        present_key_value_state = (key_states, value_states) if (self.is_decoder and use_cache) else None
+        outputs = (attn_output,) + (present_key_value_state,) + (position_bias,)
+
+        if output_attentions:
+            outputs = outputs + (attn_weights,)
+        return outputs
+
+def gaudi_MT5LayerSelfAttention_forward(
+    self,
+    hidden_states,
+    attention_mask=None,
+    position_bias=None,
+    layer_head_mask=None,
+    past_key_value=None,
+    use_cache=False,
+    output_attentions=False,
+    token_idx=None,
+):
+    """
+    Copied from https://github.com/huggingface/transformers/blob/v4.40.2/src/transformers/models/mt5/modeling_mt5.py
+    Self-attention (if key_value_states is None) or attention over source sentence (provided by key_value_states).
+    """
+
+    normed_hidden_states = self.layer_norm(hidden_states)
+    attention_output = self.SelfAttention(
+            normed_hidden_states,
+            mask=attention_mask,
+            position_bias=position_bias,
+            layer_head_mask=layer_head_mask,
+            past_key_value=past_key_value,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            token_idx=token_idx,
+        )
+    hidden_states = hidden_states + self.dropout(attention_output[0])
+    outputs = (hidden_states,) + attention_output[1:]  # add attentions if we output them
+    return outputs
+
+def gaudi_MT5Block_forward(
+        self,
+        hidden_states,
+        attention_mask=None,
+        position_bias=None,
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
+        encoder_decoder_position_bias=None,
+        layer_head_mask=None,
+        cross_attn_layer_head_mask=None,
+        past_key_value=None,
+        use_cache=False,
+        output_attentions=False,
+        return_dict=True,
+        token_idx=None,
+    ):
+        if past_key_value is not None:
+            if not self.is_decoder:

```

Comment:
```suggestion
    if past_key_value is not None:
        if not self.is_decoder:
```
Please ensure consistent indentation throughout the entire code.